---
title: "Project"
Author: "Mikey Gau, PharmD 2025"
output: html_document
date: "2024-04-09"
---

```{r setup, include=FALSE}
#knitr::opts_chunk$set(echo = TRUE)
```

## 2.How does SARS-Cov-02 infection alter transcripyional acitivity. 
### Overall design: Comparative gene expression profiling analysis of RNA-seq data using 4 types (uninfected/infected, young/mid-aged) of isolated lung endothelial cells.

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r message=FALSE, warning=FALSE, paged.print=FALSE, results="hide"}
# We first need to import the important libnrary for today's class, dplyr
library(dplyr)
library(readr)
library(DESeq2)
library(ggplot2)
library(EnhancedVolcano)

```

```{r}
library(readxl)
counts <- read.csv('C:/Users/xmike/OneDrive/Desktop/Coding_in_R/Project/counts.csv', row.names = 1, header = TRUE)
metadata <- read.csv('C:/Users/xmike/OneDrive/Desktop/Coding_in_R/Project/metacounts.csv', row.names = 1, header = TRUE)
```

```{r}
countsnew <- counts %>%
  dplyr::select(Aln_YoungPBS.sam, Aln_MidAgePBS.sam, Aln_YoungSARS.sam, Aln_MidAgeSARS.sam)
```

```{r}
### Run this chunk to import the counts table and metadata into your evironment.
#counts <- read.csv('airway_scaledcounts.csv', row.names = 1, header = TRUE)
#metadata <- read.csv('airway_metadata.csv', row.names = 1, header = TRUE)
### Modify the metadata table

sample_info <- metadata %>%
  dplyr::select_all()
sample_info


#creat a deseqobject

dds <- DESeqDataSetFromMatrix(countData = countsnew, 
                              colData = sample_info,
                              design = ~ Treatment)
#rename the SRR to the data name Aln_YoungSARS or visa versa

## Subset your ddseq object
subset <-  dds[(rowSums(counts(dds)) >= 10),]

### fix the base to a preferred treatment
relevel(dds$Treatment, ref='PBS')

### Run deseq2

deseq_ob <- DESeq(subset)

#### Save the results to a new object
res <- results(deseq_ob, alpha = 0.05)
```
```{r}
#subset
```

```{r}
#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install("org.Mm.eg.db")
```

```{r}
#BiocManager::install("EnhancedVolcano")
```

```{r}
library("org.Mm.eg.db")
sigs.df <-  as.data.frame(res)
sigs.df$symbol <- mapIds(org.Mm.eg.db, keys= rownames(sigs.df), keytype = 'ENSEMBL', colum = "SYMBOL")
```

```{r}
#keytypes(org.Mm.eg.db)
```

```{r}
#sigs.df
```
### Make your first volcano plot!
```{r}
EnhancedVolcano(sigs.df, x='log2FoldChange', y = 'padj', lab = sigs.df$symbol)

```
Look at the manual (vignette for enhanced volcano) https://bioconductor.org/packages/devel/bioc/vignettes/EnhancedVolcano/inst/doc/EnhancedVolcano.html
Create a new code chunk and plot a better version of this volcano plot! (15 minutes)

```{r}
#Write your code here!
EnhancedVolcano(sigs.df, x='log2FoldChange', y = 'padj', lab = sigs.df$symbol,col=c('black','blue','yellow','green'))+coord_flip()
```

## Heatmaps!
```{R}
#install.packages("ComplexHeatmap")
#BiocManager::install("ComplexHeatmap")
library(ComplexHeatmap)
```

```{R}
diff.df <- as.data.frame(sigs.df)
diff.df <- diff.df %>%
  filter(padj < 1)
```

```{R}
mat <- counts(deseq_ob, normalized = T)[rownames(diff.df),]
mat.z <- t(apply(mat,1, scale))
colnames(mat.z) <- colnames(mat)
Heatmap(mat.z, cluster_rows= T, cluster_columns= T, name = "Z-score")
```
## Use dplyr to subset the table of significant values until you get a nice looking figure. 
(You cannot plot all genes at once so you have to subset in order to gain some insights from 
your data)
```{R}
diff.df <- as.data.frame(sigs.df)
diff.df <- diff.df %>%
  filter(padj < 0.05, log2FoldChange > 3, baseMean > 100)

mat <- counts(deseq_ob, normalized = T)[rownames(diff.df),]
mat.z <- t(apply(mat,1, scale))
colnames(mat.z) <- colnames(mat)
Heatmap(mat.z, cluster_rows= T, cluster_columns= T, name = "Z-score", row_labels = diff.df[rownames(mat.z),]$symbol)

```
## Gene Ontology Analysis
```{R}
#BiocManager::install('clusterProfiler')
library(clusterProfiler)
```

```{R}
diff.df <- as.data.frame(sigs.df)
diff.df <- diff.df %>%
  filter(padj < 0.05, log2FoldChange >2, baseMean > 100)

genes_to_test <- rownames(diff.df)
GO_results <- enrichGO(gene = genes_to_test, OrgDb = "org.Mm.eg.db", keyType = 'ENSEMBL', ont = "BP")
fit <- plot(barplot(GO_results, showCategory=10))
```

Anothe program is gost, we can try it below.
```{R}
library(gprofiler2) 
go_upreg <- gost(genes_to_test, organism = 'mmusculus', significant = T)
gostplot(go_upreg, interactive = T)
```

It might be more informative to create plots for genes that are both upregulated and downregulated. Subset your data above and plot the GO enrichment for both upregulated and downreghulated.

GSEA Plot, make one, Cesar will explain what it means 
```{R}
res <- res[order(-res$stat),]
gene_list <- res$stat
names(gene_list) <- rownames(res)
gse <- gseGO(gene_list, ont = "BP", keyType = "ENSEMBL", OrgDb ="org.Mm.eg.db", eps = 1e-300 )
gseaplot(gse, geneSetID = "GO:0046324" )

```
